crypto-core/src/main/java/com/palantir/crypto2/io/ApacheCtrDecryptingSeekableInput.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.crypto2.io;
+
+import com.palantir.crypto2.keys.KeyMaterial;
+import com.palantir.seekio.SeekableInput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Properties;
+import org.apache.commons.crypto.Crypto;
+import org.apache.commons.crypto.cipher.CryptoCipherFactory;
+import org.apache.commons.crypto.stream.CtrCryptoInputStream;
+import org.apache.commons.crypto.stream.input.Input;
+import org.apache.commons.crypto.utils.Utils;
+
+/**
+ * A {@link SeekableInput} that decrypts AES/CTR encrypted SeekableInputs using the given {@link KeyMaterial}. This
+ * implementation uses Apache's {@link CtrCryptoInputStream} which uses OpenSSL and supports AES-NI.
+ */
+public final class ApacheCtrDecryptingSeekableInput extends CtrCryptoInputStream implements SeekableInput {
+
+    private static final String ALGORITHM = "AES/CTR/NoPadding";
+    private static final int BUFFER_SIZE = 8192;
+    private static final Properties PROPS;
+
+    static {
```

Comment:
generally prefer you make this a static method and then call:

    ... PROPS = initializeProps();